### PR TITLE
imagestreamtag race

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/core.py [hunk 1]
```diff
@@ -841,6 +842,16 @@ def get_image_stream_tag(self, tag_id):
         check_response(response, log_level=logging.DEBUG)
         return response
 
+    @retry_on_not_found
+    def get_image_stream_tag_with_retry(self, tag_id):
```
Comment (twaugh) [inline osbs/core.py]: We don't have any test coverage for this -- is this ok?

File: osbs/utils.py [hunk 2]
```diff
@@ -702,6 +702,19 @@ def should_retry_cb(ex):
     return retry
 
 
+def retry_on_not_found(func):
+    @wraps(func)
+    def retry(*args, **kwargs):
+        # Only retry when OsbsResponseException was raised due to not found
+        def should_retry_cb(ex):
+            return ex.status_code == http_client.NOT_FOUND
```
Comment (twaugh) [inline osbs/utils.py]: We don't have coverage here according to coveralls, but we do on the equivalent part of `retry_on_conflict`.

File: osbs/core.py (hunk 1 above)
Comment (rcerven) [reply osbs/core.py]: that should be covered by https://github.com/containerbuildsystem/osbs-client/pull/911/files#diff-625efbbf35a2290861f596fee1b19838R2172

not sure why it is showing that it isn't

File: osbs/utils.py (hunk 2 above)
Comment (rcerven) [reply osbs/utils.py]: I will try to add it , I think I have idea

Comment (rcerven) [reply osbs/utils.py]: idea failed

File: osbs/core.py (hunk 1 above)
Comment (rcerven) [reply osbs/core.py]: nvm that test is testing method in api, and we are always mocking methods in core

**Comment by rcerven** [comment]
rn updated: - prevent tag race condition when updating imagestreamtag for autorebuilds in multiple builds at the same time

**Pull request merged**